### PR TITLE
Bug 1130742 - keyboard_test.js will fail if js/layouts contains untracked non-JS file r=timdream

### PR DESCRIPTION
**Pull request by Anindya-Pandey**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[Anindya-Pandey Anindya-Pandey](https://github.com/Anindya-Pandey) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=9e68c0d41151)

File: apps/keyboard/test/build/integration/keyboard_test.js [hunk 1]
```diff
@@ -100,10 +100,18 @@ suite('Keyboard layouts building tests', function() {
...
-        return 'js/layouts/' + layout + '.js';
-      });
+      
+     var layoutIds1 = [];
+     layoutIds1 = layoutIds.filter(function(layout) {
+     	if(layout)
+     		return layout;
+     });
+     
+     layoutIds = layoutIds1;
```
Comment (timdream) [inline apps/keyboard/test/build/integration/keyboard_test.js]: You can chain these arrays with `var layouts = layoutIds.filter(...).map(...);`

Where the `filter()` function will simply return the array (the one you named `layoutIds1`) implicitly for you to call (`map()`), and `layouts` will be the value returning from `map()`.

File: apps/keyboard/test/build/integration/keyboard_test.js [hunk 2]
```diff
@@ -100,10 +100,18 @@ suite('Keyboard layouts building tests', function() {
...
         });
-
-      var layouts = layoutIds.map(function(layout) {
-        return 'js/layouts/' + layout + '.js';
-      });
+      
+     var layoutIds1 = [];
+     layoutIds1 = layoutIds.filter(function(layout) {
+     	if(layout)
+     		return layout;
```
Comment (timdream) [inline apps/keyboard/test/build/integration/keyboard_test.js]: The callback function you set up here is invalid -- this will not filter out anything.

Comment (timdream) [inline apps/keyboard/test/build/integration/keyboard_test.js]: The correct array to filter is the array returned by `fs.readdirSync(appDirPath + '/js/layouts')` -- we need to use a `filter()` here and filter out file names not ended with `.js`.

**Comment by try-server-hook** [comment]
[Anindya-Pandey Anindya-Pandey](https://github.com/Anindya-Pandey) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=dac076dc0fcc)

**Comment by try-server-hook** [comment]
[Anindya-Pandey Anindya-Pandey](https://github.com/Anindya-Pandey) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5e45aa1fbba4)

File: apps/keyboard/test/build/integration/keyboard_test.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ suite('Keyboard layouts building tests', function() {
       var layouts = layoutIds.map(function(layout) {
         return 'js/layouts/' + layout + '.js';
       });
-
+
```
Comment (timdream) [inline apps/keyboard/test/build/integration/keyboard_test.js]: Please configure your editor to remove the white spaces.

File: apps/keyboard/test/build/integration/keyboard_test.js [hunk 4]
```diff
@@ -93,17 +93,15 @@ suite('Keyboard layouts building tests', function() {
...
+          return path.extname(filename) === '.js';
+        }).map(function(filename) {
           return path.basename(filename, '.js');
         });
-
-      var layouts = layoutIds.map(function(layout) {
-        return 'js/layouts/' + layout + '.js';
-      });
+     
+     var layouts = layoutIds.map(function(layout) {
```
Comment (timdream) [inline apps/keyboard/test/build/integration/keyboard_test.js]: Indent error here.

File: apps/keyboard/test/build/integration/keyboard_test.js [hunk 5]
```diff
@@ -93,17 +93,15 @@ suite('Keyboard layouts building tests', function() {
         'webapps', 'keyboard.gaiamobile.org', 'application.zip');
       var appDirPath = config.GAIA_DIR + '/apps/keyboard';
       var layoutIds =
-        fs.readdirSync(appDirPath + '/js/layouts').map(function(filename) {
-          if (path.extname(filename) !== '.js') {
-            return;
-          }
-
+        fs.readdirSync(appDirPath + '/js/layouts').filter(function(filename) {
+          return path.extname(filename) === '.js';
```
Comment (timdream) [inline apps/keyboard/test/build/integration/keyboard_test.js]: `return (path.extname(filename) === '.js');`

**Pull request merged**